cfgmgr/timerangemgr.cpp

```diff
@@ -8,15 +8,31 @@
 #include "timerangemgr.h"
 #include "tokenize.h"
 #include <table.h>
+#include "croncpp.h"
 
 using namespace std;
 using namespace swss;
+using namespace cron;
 
 TimeRangeMgr::TimeRangeMgr(DBConnector *cfgDb, DBConnector *stateDb, const vector<string> &tableNames) : Orch(cfgDb, tableNames),
                                                                                                          m_stateTimeRangeStatusTable(stateDb, STATE_TIME_RANGE_STATUS_TABLE_NAME)
 {
 }
 
+
+bool is_time_in_range(const cronexpr& startExpr, const cronexpr& endExpr, const std::tm& currentTM) {
+    std::time_t currentTime = mktime(const_cast<tm*>(&currentTM)); // Convert currentTM to time_t
+    time_t previousMinute = currentTime - 60; // Go back one minute to ensure full coverage
+
```

Comment:
Doesn't this add risk of getting an "event" for something that you already activate here? do you handle such case?